Content.Server/Weapon/Melee/Esword/EswordSystem.cs

```diff
@@ -0,0 +1,135 @@
+    {
+        [Dependency] private readonly ActionBlockerSystem _blockerSystem = default!;
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<EswordComponent, ComponentInit>(OnComponentInit);
+            SubscribeLocalEvent<EswordComponent, MeleeHitEvent>(OnMeleeHit);
+            SubscribeLocalEvent<EswordComponent, UseInHandEvent>(OnUseInHand);
+            SubscribeLocalEvent<EswordComponent, InteractUsingEvent>(OnInteractUsing);
+        }
+
+        private void OnComponentInit(EntityUid uid, EswordComponent comp, ComponentInit args)
+        {
+            string[] possibleColors = { "Tomato", "DodgerBlue", "Aqua", "MediumSpringGreen", "MediumOrchid" };
+            Random random = new Random();
+            comp.BladeColor = Color.FromName(possibleColors[random.Next(5)]);
+        }
+
+        private void OnMeleeHit(EntityUid uid, EswordComponent comp, MeleeHitEvent args)
+        {
+            
+            if (comp.Activated == true)
+            {
+                args.BonusDamage.DamageDict = new Dictionary<string, FixedPoint2>()
+                {
+                    {"Slash", FixedPoint2.New(25)},
+                };
+                args.HitSoundOverride = comp.HitSound;
+            }
+        }
+
+        private void OnUseInHand(EntityUid uid, EswordComponent comp, UseInHandEvent args)
+        {
+            if (!_blockerSystem.CanUse(args.User))
+                return;
+
+            if (comp.Activated)
+            {
+                TurnOff(comp);
+            }
+            else
+            {
+                TurnOn(comp, args.User);
+            }
+        }
+
+        private void TurnOff(EswordComponent comp)
+        {
+            if (!comp.Activated)
+                return;
+
+            if (!TryComp(comp.Owner, out SpriteComponent? sprite)
+                || !TryComp(comp.Owner, out SharedItemComponent? item))
+                return;
+
+            item.Size = 5;
+
+            SoundSystem.Play(Filter.Pvs(comp.Owner), comp.DeActivateSound.GetSound(), comp.Owner);
+
+            item.EquippedPrefix = "off";
+            sprite.LayerSetVisible(1, false);
+
+            comp.Activated = false;
+        }
+
+        private void TurnOn(EswordComponent comp, EntityUid user)
+        {
+            if (comp.Activated)
+                return;
+
+            if (!TryComp(comp.Owner, out SpriteComponent? sprite)
+                || !TryComp(comp.Owner, out SharedItemComponent? item))
+                return;
+
+            item.Size = 9999;
+
+            SoundSystem.Play(Filter.Pvs(comp.Owner), comp.ActivateSound.GetSound(), comp.Owner);
+
+            if (comp.Hacked == true)
+            {
+                item.EquippedPrefix = "on-rainbow";
+                sprite.LayerSetState(0, "e_sword_rainbow_on");
+            }
+            else
+            {
+                item.EquippedPrefix = "on";
+                item.Color = comp.BladeColor;
```

Comment:
You need to set `item.Color = Color.White` somewhere for hacked blades, as otherwise the rainbow-effect blades are all still tinted by the original colour.